Chapter4/section_one.rst

```diff
@@ -0,0 +1,121 @@
+to understand how to clone a repository on GitHub.
+
+Next, we go through the library and find the raw dataset that we want to
+process, in this case, the ``HarmonizedWorldSoilDatabase`` dataset. It
+is important to note that the ``HarmonizedWorldSoilDatabase`` is a test
+dataset. Each dataset will have specific transformation steps to prepare
+analysis, but each step can be run using our DVC pipelines. Each
+pipeline will follow the broad steps of download, extract and transform,
+then save the results on the remote storage.
+
+In the ``Land_Sector_Datasets`` repository we just cloned, navigate to
+the ``bilToTif.py`` file, which is nested in the ``Data/Soil``
+directory.
+
+This ``Data/Soil/bilToTif.py`` file contains a ``BILtoTIF`` function
+that takes the raw ``HarmonizedWorldSoilDatabase`` dataset as a
+parameter and then converts the dataset from a ``.bil`` file to a
+``.tif`` file. The ``BILtoTIF`` function then saves the ``.tif`` file.
+
+.. code:: py
+
+      def BILtoTIF(inBilPath,outTifPath):
+          inBil = gdal.Open(inBilPath)
+          driver = gdal.GetDriverByName('GTiff')
+          outTif = driver.CreateCopy(outTifPath, inBil, 0)
+    
+          inBil = None
+          outTif = None
+
+Next, in the ``Data/Soil/bilToTif.py`` file, we have a
+``restructureTIF`` function that restructures the ``.tif`` dataset using
+the options specified in the ``restructureTIF``.
+
+.. code:: py
+
+      def restructureTIF(in_tif, out_tif):
+            options = gdal.WarpOptions(
+                creationOptions=["COMPRESS=DEFLATE","PREDICTOR=2","ZLEVEL=9"],
+                dstSRS="EPSG:4326",
+                format="GTiff",
+                multithread=True,
+                xRes=0.005, yRes=0.005,
+                resampleAlg="near",
+             )
+         gdal.Warp(destNameOrDestDS=out_tif, srcDSOrSrcDSTab=in_tif, options=options)
+   if __name__ == "__main__":
+      in_src = os.path.join("HWSD_RASTER","hwsd.bil")
+      out_src = os.path.join("HWSD_VECTOR","HarmonizedWorldSoilDatabase_RAW.tif")
+      BILtoTIF(in_src,out_src)
+      if os.path.isfile(out_src):
+          restructuredTifPath = os.path.join("HWSD_VECTOR","HarmonizedWorldSoilDatabase_RESTRUCTURED.tif")
+          restructureTIF(in_tif=out_src, out_tif=restructuredTifPath)
+
+We can see the different options specified in the code block above.
+
+The project’s root directory has a ``dvc.yaml`` file. This ``dvc.yaml``
+file contains an established DVC pipeline.
+
+.. code:: py
+
+
+    vars:
+       - python: C:\Develop\anaconda\envs\gdal\python.exe
+       stages:
+           HarmonizedWorldSoilDatabase:
+               cmd: python bilToTif.py
+               wdir: Data/Soil
+               deps:
+               - HWSD_RASTER\hwsd.bil
+               - bilToTif.py
+               outs:
+               - HWSD_VECTOR\HarmonizedWorldSoilDatabase_RAW.tif
+               - HWSD_VECTOR\HarmonizedWorldSoilDatabase_RESTRUCTURED.tif
+
+
+The DVC pipeline does the following: 
```

Comment:
I think it's worth putting the DVC pipeline first, then explaining the `bilToTif` script.